pkgs/development/tools/jdt-ls/jdt-ls-wrapper.sh

```diff
@@ -0,0 +1,73 @@
+#!@shell@
+
+# This script is a wrapper around jdt.ls. It sets some options to the only
+# valid value, while it defines some default values for other options, but
+# provides a way to override those defaults.
+# Any additional unknown args will be appended to the command  (e.g. "-Xdiag")
+# Usage example:
+# ${pkgs.jdt-ls}/bin/jdt-ls -data /some/path -Xdiag
+
+# Documentation about the parameters:
+# https://github.com/eclipse/eclipse.jdt.ls/wiki/Running-the-JAVA-LS-server-from-the-command-line
+# Some of the values are the recommended by the nvim client:
+# https://github.com/mfussenegger/nvim-jdtls#language-server-installation
+
+# Log and config directories are mixed, see
+# https://github.com/eclipse/eclipse.jdt.ls/issues/1097
+# The config.ini file needs to be in this directory, logs are written here too.
+# We use a temp directory because the language server logs are usually not
+# important to the end user
+# This directory is deleted when the script stops
+mkdir -p /tmp/jdtls_config
+configuration="$(mktemp -d /tmp/jdtls_config/XXXXX)"
+trap "{ rm -rf $configuration; }" EXIT
+
+log_level="-Dlog.level=INFO"
+xms="-Xms1g"
+xmx="-Xmx2G"
+
+# Data directory (also called workspace directory in jdt.ls docs) is used as
+# some kind of cache by jdt.ls. For that reason, we want to use different
+# directories per project. We use the basename of $PWD
+# Data directory CANNOT be in the project itself, it needs to be in a different
+# directory
+data="${XDG_CACHE_HOME:-$HOME/.cache}/jdtls/${PWD##*/}"
+
+params=("$@")
+extra_args=()
+for ((n = 0; n < ${#params[*]}; n += 1)); do
+    p="${params[$n]}"
+    if [[ "$p" == "-Dlog.level="* ]]; then
+        log_level="$p"
+    elif [[ "$p" == "-Xms"* ]]; then
+        xms="$p"
+    elif [[ "$p" == "-Xmx"* ]]; then
+        xmx="$p"
+    elif [[ "$p" == "-configuration" ]]; then
+        configuration="${params[$((n + 1))]}"
+        n=$((n + 1))
+    elif [[ "$p" == "-data" ]]; then
+        data="${params[$((n + 1))]}"
+        n=$((n + 1))
+    else
+        extra_args+=("$p")
+    fi
+done
+
+cp @config_path@/config.ini "$configuration"
+
+# use java version from PATH, @jdk@ as fallback
+java_bin=$(command -v java || echo "@jdk@")
+$java_bin \
+    -Declipse.application=org.eclipse.jdt.ls.core.id1 \
+    -Dosgi.bundles.defaultStartLevel=4 \
+    -Declipse.product=org.eclipse.jdt.ls.core.product \
+    -jar @launcher@ \
+    --add-modules=ALL-SYSTEM \
+    --add-opens java.base/java.util=ALL-UNNAMED \
+    --add-opens java.base/java.lang=ALL-UNNAMED \
+    "$log_level" \
```

Comment:
I think -D and -X options need to come before `-jar` to take effect.

Reply:
You are right, fixing it